flocker/control/_persistence.py

```diff
@@ -87,7 +93,58 @@ def decode_object(dictionary):
                        [_DEPLOYMENT_FIELD], [])
 
 
-class ConfigurationPersistenceService(Service):
+class LeaseService(Service):
+    """
+    Manage leases.
+    In particular, clear out expired leases once a second.
+
+    :ivar _reactor: A ``twisted.internet.reactor`` implementation.
+    :ivar _persistence_service: The persistence service to act with.
+    :ivar _lc: A ``twisted.internet.task.LoopingCall`` run every second
+        to update the configured leases by releasing leases that have
+        expired.
+    """
+    def __init__(self, reactor, persistence_service):
+        if reactor is None:
+            reactor = default_reactor
+        self._reactor = reactor
+        self._persistence_service = persistence_service
+
+    def startService(self):
+        self._lc = LoopingCall(self._expire)
+        self._lc.clock = self._reactor
+        self._lc.start(1)
+
+    def stopService(self):
+        self._lc.stop()
+
+    def _expire(self):
+        now = datetime.now(tz=UTC)
+        return update_leases(lambda leases: leases.expire(now),
+                             self._persistence_service)
```

Comment:
Thanks for the added test. Could you please update the `XXX` comment in the issue below with the exact ticket for when this will be actually used (rather than the higher-level ticket)? If you can do that, then please feel free to merge. If it's not possible for some reason, then please bounce back to me.